editor/editor_properties.cpp

```diff
@@ -790,262 +790,293 @@ EditorPropertyFlags::EditorPropertyFlags() {
+	if (read_only) {
+		return;
+	}
+	const Ref<InputEventMouseMotion> mm = p_ev;
+	if (mm.is_valid()) {
+		bool expand_was_hovered = expand_hovered;
+		expand_hovered = expand_rect.has_point(mm->get_position());
+		if (expand_hovered != expand_was_hovered) {
+			update();
+		}
+
+		if (!expand_hovered) {
+			for (int i = 0; i < flag_rects.size(); i++) {
+				if (flag_rects[i].has_point(mm->get_position())) {
+					// Used to highlight the hovered flag in the layers grid.
+					hovered_index = i;
+					update();
+					return;
+				}
 			}
 		}
-		return String();
-	}
 
-	void gui_input(const Ref<InputEvent> &p_ev) override {
-		if (read_only) {
-			return;
+		// Remove highlight when no square is hovered.
+		if (hovered_index != -1) {
+			hovered_index = -1;
+			update();
 		}
-		const Ref<InputEventMouseMotion> mm = p_ev;
-		if (mm.is_valid()) {
-			bool expand_was_hovered = expand_hovered;
-			expand_hovered = expand_rect.has_point(mm->get_position());
-			if (expand_hovered != expand_was_hovered) {
-				update();
-			}
 
-			if (!expand_hovered) {
-				for (int i = 0; i < flag_rects.size(); i++) {
-					if (flag_rects[i].has_point(mm->get_position())) {
-						// Used to highlight the hovered flag in the layers grid.
-						hovered_index = i;
-						update();
-						return;
-					}
-				}
-			}
+		return;
+	}
 
-			// Remove highlight when no square is hovered.
-			if (hovered_index != -1) {
-				hovered_index = -1;
-				update();
+	const Ref<InputEventMouseButton> mb = p_ev;
+	if (mb.is_valid() && mb->get_button_index() == MouseButton::LEFT && mb->is_pressed()) {
+		if (hovered_index >= 0) {
+			// Toggle the flag.
+			// We base our choice on the hovered flag, so that it always matches the hovered flag.
+			if (value & (1 << hovered_index)) {
+				value &= ~(1 << hovered_index);
+			} else {
+				value |= (1 << hovered_index);
 			}
 
-			return;
+			emit_signal(SNAME("flag_changed"), value);
+			update();
+		} else if (expand_hovered) {
+			expanded = !expanded;
+			update_minimum_size();
+			update();
 		}
-
-		const Ref<InputEventMouseButton> mb = p_ev;
-		if (mb.is_valid() && mb->get_button_index() == MouseButton::LEFT && mb->is_pressed()) {
-			if (hovered_index >= 0) {
-				// Toggle the flag.
-				// We base our choice on the hovered flag, so that it always matches the hovered flag.
-				if (value & (1 << hovered_index)) {
-					value &= ~(1 << hovered_index);
-				} else {
-					value |= (1 << hovered_index);
-				}
-
-				emit_signal(SNAME("flag_changed"), value);
-				update();
-			} else if (expand_hovered) {
-				expanded = !expanded;
-				update_minimum_size();
-				update();
-			}
+	}
+	if (mb.is_valid() && mb->get_button_index() == MouseButton::RIGHT && mb->is_pressed()) {
+		if (hovered_index >= 0) {
+			renamed_layer_index = hovered_index;
+			layer_rename->set_position(get_screen_position() + mb->get_position());
+			layer_rename->set_size(Vector2(1, 1));
```

Comment:
If this is meant to force resetting the size to minsize, you can use `layer_rename->reset_size();` instead.